src/QuadForm/Herm/Lattices.jl

```diff
@@ -750,3 +750,92 @@ function maximal_integral_lattice(V::HermSpace)
   return maximal_integral_lattice(L)
 end
 
+################################################################################
+#
+#  Intersection
+#
+################################################################################
+
+function _reconstruction_herm_lattice(M, f)
+  Vas = codomain(f)
+  E = base_ring(Vas)
+  OE = maximal_order(E)
+  n = absolute_degree(E)
+
+  K = base_field(E)
+  OK = maximal_order(K)
+  @assert divides(ncols(M), n)[1]
+  deg = divexact(ncols(M),n)
+  
+  a = zero_matrix(E, deg, deg)
+  coeffs = (Hecke.fractional_ideal_type(OE))[]
+  
+  for i = 1:deg
+    ind = findfirst(k -> !iszero(M[k,:]), 1+(i-1)*n:i*n) + (i-1)*n
+    if typeof(ind) === nothing
+      error("Zero block detected.")
+    end
+    v = [M[ind, j]  for j = 1:ncols(M)] 
+    a[i, :] = f(v)  
+  
+    index = findfirst(k -> !iszero(a[i, k]), 1:deg) 
+    A = M[(1+(i-1)*n):i*n, (1+(index-1)*n): index*n] 
+    A_E = _divide_Qmat_by_nf_elem(A, a[i, index], E)
+    y = [_Qmat_to_Evec(A_E[1, :], E)[1], _Qmat_to_Evec(A_E[degree(K)+1, :], E)[1]]
+    A_E[1:degree(K),:] = _divide_Qmat_by_nf_elem(A_E[1:degree(K), :], y[1], E)
+    A_E[degree(K)+1:n,:] = _divide_Qmat_by_nf_elem(A_E[degree(K)+1:n, :], y[2], E)
+    
+    if !iszero(A_E[:,degree(K)+1:end]) 
+      error("The lattice cannot be lifted.")
+    end
+    
+    fk1 = FakeFmpqMat(A_E[1:degree(K), 1:degree(K)])
+    frac_ideal1 = fractional_ideal(OK, fk1)
+    B1 = fractional_ideal(OE, frac_ideal1)
+    
+    fk2 = FakeFmpqMat(A_E[degree(K)+1:n, 1:degree(K)])
+    frac_ideal2 = fractional_ideal(OK, fk2)
+    B2 = fractional_ideal(OE, frac_ideal2)
+
+    Ai = B1*y[1] + B2*y[2]
+    push!(coeffs, Ai)
+  end 
+  
+  pmatrix = Hecke.PMat{Hecke.elem_type(E), Hecke.fractional_ideal_type(OE)}(a, coeffs)
+  Lnew = lattice(Vas, pmatrix) 
+  return Lnew
+end
+
+@doc Markdown.doc"""
+    intersect_herm_lattice(M::HermLat, N::HermLat) -> HermLat
+
+  Given two hermitian lattices `M` and `N`, return their intersection
+  using restriction of scalars.
+  
+  The lattices `M` and `N` must have the same ambient space.
+"""
+function intersect_herm_lattice(M::HermLat, N::HermLat)
+
+  @req ambient_space(M) === ambient_space(N) "Lattices must have the same ambient space"
+
+  is_sublattice(M, N) && return N
+  is_sublattice(N, M) && return M
+
+  MM, f = restrict_scalars_with_map(M)
+  NN = restrict_scalars_with_respect_to_map(N, f)
+
+  BM = basis_matrix(MM)
+  BN = basis_matrix(NN)
+  dM = denominator(BM)
+  dN = denominator(BN)
+  d = lcm(dM, dN)
+  BMint = change_base_ring(FlintZZ, d * BM)
+  BNint = change_base_ring(FlintZZ, d * BN)
+  H = vcat(BMint, BNint)
+  k, K = left_kernel(H)
+  BI = divexact(change_base_ring(FlintQQ, view(K, 1:k, 1:nrows(BM)) * BMint), d)
+  LLint = lattice(ambient_space(MM), BI)
+
+  return _reconstruction_herm_lattice(basis_matrix(LLint), f)
```

Comment:
```suggestion
  BI = divexact(change_base_ring(FlintQQ, view(K, 1:k, 1:nrows(BM)) * BMint), d)

  return _reconstruction_herm_lattice(BI,  f)
```
Indeed you don't need to create the lattice and then access its basis matrix, already use the basis matrix you have constructed